### PR TITLE
Support for Kaleidoscope Horizontal Mouse Scroll Wheel

### PR DESCRIPTION
**Pull request by SjB**

### CODE REVIEW

**Comment by algernon** [comment]
In general, we try to avoid changing things in a backward-incompatible way, in order not to break user sketches, and third-party plugins. Would it be possible to update this PR so that it keeps the old variant of the changed functions? They could just call the 4-argument variant with the new wheel direction set to 0.

That way, we'd both preserve compatibility, and introduce the new feature!

I expect that the existing wheel will be used more often, so I wouldn't deprecate the 3-arg variants either.

Other than this, the change looks good, thank you!

**Comment by obra** [comment]
It may make more sense to redefine the 3arg form as a call to the 4 arg form. that gives us less code that can drift apart.

**Comment by algernon** [comment]
> It may make more sense to redefine the 3arg form as a call to the 4 arg form. that gives us less code that can drift apart.

That's what I was trying to say too. `ENOCOFFEEYET` happened O:)

**Comment by algernon** [comment]
Depends on keyboardio/KeyboardioHID#12.

File: src/kaleidoscope/hid.cpp [hunk 1]
```diff
@@ -135,8 +135,13 @@ void initializeMouse() {
 
 void moveMouse(signed char x, signed char y, signed char wheel) {
   Mouse.move(x, y, wheel);
+  //moveMouseWithHorizontalWheel(x, y, wheel, 0);
 }
 
+void moveMouseWithHorizontalWheel(signed char x, signed char y, signed char vWheel, signed char hWheel) {
```
Comment (algernon) [inline src/kaleidoscope/hid.cpp]: I believe we could combine `moveMouse` and `moveMouseWithHorizontalWheel`, like this:

```c++
void moveMouse(signed char x, signed char y, signed char vWheel = 0, signed char hWheel = 0) {
  Mouse.move(x, y, vWheel, hWheel);
}
```

With the default values for the `vWheel` and `hWheel` arguments, `moveMouse(x, y)` and `moveMouse(x, y, vWheel)` will continue working as before.

No API breakage, no new functions introduced!

File: src/kaleidoscope/hid.cpp [hunk 2]
```diff
@@ -156,10 +161,19 @@ void initializeAbsoluteMouse() {
...
 }
+
 void moveAbsoluteMouseTo(uint16_t x, uint16_t y, signed char wheel) {
-  SingleAbsoluteMouse.moveTo(x, y, wheel);
+  moveAbsoluteMouseWithHorizontalWheelTo(x, y, wheel, 0);
+}
+
+void moveAbsoluteMouseWithHorizontalWheelTo(uint16_t x, uint16_t y, signed char vWheel, signed char hWheel) {
+  SingleAbsoluteMouse.moveTo(x, y, vWheel, hWheel);
 }
```
Comment (algernon) [inline src/kaleidoscope/hid.cpp]: See my previous note, these functions can be combined into one while still preserving the old API, if you use default arguments. Thus, only `moveAbsoluteMouse(x, y, vWheel = 0, hWheel = 0)` and `moveAbsoluteMouseTo(x, y, vWheel = 0, hWheel = 0)` would remain.

**Comment by obra** [comment]
Awesome. Would you mind squashing all the commits together?

**Comment by algernon** [comment]
We can also do a squash merge on GitHub. Not sure if that lets us edit the commit message, though.

**Comment by algernon** [comment]
Closing this one in favour of #256.

**Pull request closed without merging**